src/api_proxy/service_control/check_response_test.cc

```diff
@@ -29,174 +29,236 @@ namespace {
+  CheckResponseInfo info;
+  Status result =
+      ConvertCheckErrorToStatus(CheckError::BILLING_DISABLED, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::CONSUMER_ERROR);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithSecurityPolicyViolated) {
+  CheckResponseInfo info;
   Status result =
-      ConvertCheckErrorToStatus(CheckError::SECURITY_POLICY_VIOLATED);
+      ConvertCheckErrorToStatus(CheckError::SECURITY_POLICY_VIOLATED, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::CONSUMER_BLOCKED);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithInvalidCredentail) {
-  Status result = ConvertCheckErrorToStatus(CheckError::INVALID_CREDENTIAL);
+  CheckResponseInfo info;
+  Status result =
+      ConvertCheckErrorToStatus(CheckError::INVALID_CREDENTIAL, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::CONSUMER_ERROR);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithLocationPolicyViolated) {
+  CheckResponseInfo info;
   Status result =
-      ConvertCheckErrorToStatus(CheckError::LOCATION_POLICY_VIOLATED);
+      ConvertCheckErrorToStatus(CheckError::LOCATION_POLICY_VIOLATED, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::CONSUMER_BLOCKED);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithConsumerInvalid) {
-  Status result = ConvertCheckErrorToStatus(CheckError::CONSUMER_INVALID);
+  CheckResponseInfo info;
+  Status result =
+      ConvertCheckErrorToStatus(CheckError::CONSUMER_INVALID, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::CONSUMER_ERROR);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithResourceExhuasted) {
-  Status result = ConvertCheckErrorToStatus(CheckError::RESOURCE_EXHAUSTED);
+  CheckResponseInfo info;
+  Status result =
+      ConvertCheckErrorToStatus(CheckError::RESOURCE_EXHAUSTED, &info);
   EXPECT_EQ(Code::RESOURCE_EXHAUSTED, result.code());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::ERROR_TYPE_UNSPECIFIED);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithAbuserDetected) {
-  Status result = ConvertCheckErrorToStatus(CheckError::ABUSER_DETECTED);
+  CheckResponseInfo info;
+  Status result = ConvertCheckErrorToStatus(CheckError::ABUSER_DETECTED, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
-  EXPECT_EQ("", result.error_message());
+  EXPECT_EQ(info.error_type, CheckResponseErrorType::CONSUMER_ERROR);
 }
 
 TEST(CheckResponseTest, WhenResponseIsBlockedWithApiTargetBlocked) {
-  Status result = ConvertCheckErrorToStatus(CheckError::API_TARGET_BLOCKED);
+  CheckResponseInfo info;
+  Status result =
+      ConvertCheckErrorToStatus(CheckError::API_TARGET_BLOCKED, &info);
   EXPECT_EQ(Code::PERMISSION_DENIED, result.code());
```

Comment:
Can we also make a `runTest` for this part?

Reply:
Done. 

I wonder if these tests are actually needed. In fact, the mappings defined in the test code could just be used in the production code to replace the switch/case, would be easier to read :)